python/cudf/cudf/core/index.py

```diff
@@ -1640,27 +1690,43 @@ def __repr__(self):
             preprocess = concat([top, bottom])
         else:
             preprocess = self
-        if preprocess._values.nullable:
-            output = (
-                self.__class__(preprocess._values.astype("O").fillna("null"))
-                .to_pandas()
-                .__repr__()
-            )
+
+        if isinstance(preprocess, CategoricalIndex):
+            output = preprocess.to_pandas().__repr__()
+            output = output.replace("nan", "null")
+        elif preprocess._values.nullable:
+            output = self._clean_nulls_from_index().to_pandas().__repr__()
```

Comment:
here as well

Reply:
Same here, `Index.to_string` is not there yet in pandas.